### ISSUE TITLE
how to output the loss of a single input image?

### ISSUE THREAD

**Issue by dplgehgb** [rbgirshick/py-faster-rcnn#627]
Given one image as the input, and given the labels for objects in the image, how can I calculate the loss function by perform the forward pass?

**Comment by eugene123tw** [comment]
Have you try to run predict with only one image?

**Comment by dplgehgb** [comment]
I try to run predict with only one image, but without the ground truth label, the loss function should not be correct

**Comment by dplgehgb** [comment]
I guess I should pass in ground truth label using:
    net.blobs['gt_boxes'].data[...] = ground truth label
and pass image using:
    net.blobs['data'].data[...] = image_in
is this correct?

**Comment by eugene123tw** [comment]
You can download VOCdevkit and checkout the structure of the groud truth (```gt_boxes```) annotation xml. 
Then follow the instructions on the README.md section ```Beyond the demo: installation for training and testing models```. I think you will know better if you try to implement this. Cheers.

**Comment by dplgehgb** [comment]
Thanks. However, the part you suggested does not contain the part of calculating the loss given one single input image. The matlab version of faster-rcnn has an explicit function check_loss that can calculate the loss given a single image. In this py-faster-rcnn version, it should have this part since it will need the loss function to perform back propagation. However, caffe may put this part inside the c++ code and just implicitly calculate the loss function. I'm still trying to figure out how to do this with python.

**Comment by zhenni** [comment]
@dplgehgb 
For easy implementation, you can just using only one image index in `imageset` file.
A better way is that you can write your own code according to `lib/fast_rcnn/test.py`. Main modification should be in function `im_detect`. BTW, take a notice of the resize ratio of input image, you may also need to resize your input `gt_boxes`
